src/System.IO.Packaging/src/System/IO/Packaging/PackUriHelper.cs

```diff
@@ -448,14 +446,13 @@ public static Uri GetRelationshipPartUri(Uri partUri)
             // Get the parname without the last segment
             partName = partName.Substring(0, partName.Length - file.Length);
 
-            partName = Path.Combine(partName, s_relationshipPartSegmentName);         // Adding the "_rels" segment
-            partName = Path.Combine(partName, file);                                 // Adding the last segment back
-            partName = String.Concat(partName, s_relationshipPartExtensionName);      // Adding the ".rels" extension
+            partName = Path.Combine(partName, s_relationshipPartSegmentName, file); // Adding the "_rels" segment and the last segment back
+            partName += s_relationshipPartExtensionName;                            // Adding the ".rels" extension
             partName = partName.Replace(BackwardSlashChar, ForwardSlashChar);
```

Comment:
Should this replacement be done efore appending s_relationshipPartExtensionName?  If we know that extension name doesn't contain back slashes, then we could avoid the (small amount of work) in Replace looking at characters we know won't need to be replaced, plus if any change needs to be made, we'll be allocating slightly less memory.